draft-ietf-ppm-dap.md

```diff
@@ -512,12 +513,42 @@ implies HTTP status code 400 Bad Request unless explicitly specified otherwise.
 
 # Protocol Definition
 
-DAP has three major interactions which need to be defined:
+DAP is structured as an HTTP application consisting of three major interactions
+or sub-protocols:
 
-* Uploading reports from the client to the aggregators, specified in
-  {{upload-flow}}
-* Computing the results of a given measurement, specified in {{aggregate-flow}}
-* Collecting aggregated results, specified in {{collect-flow}}
+* Uploading reports from the client to the aggregators, using the HPKE
+  configuration ({{hpke-config-resource}}) and report ({{report-resource}})
+  resources, as described in {{upload-flow}}.
+* Computing the aggregate over sets of reports, using the aggregation job
+  resource ({{aggregation-job-resource}}), as described in {{aggregate-flow}}.
+* Collecting aggregation results, using the aggregate share
+  ({{aggregate-share-resource}}) and collection ({{collection-resource}})
+  resources, as described in {{collect-flow}}.
+
+In this section, we discuss each of the resources and the messages used to act
+on these resources. A resource's path is resolved relative to a server's
+endpoint to construct a resource URI. A server's endpoint MAY contain
+arbitrarily many path components before the DAP resources.
+
+Resource paths are specified as templates like:
+
+~~~
+/resource_type/{resource-id}
+~~~
+
+Templates may also include optional query parameters, indicated by square
+brackets ("[" and "]"), as in:
+
+~~~
+/resource_type/{resource-id}[?query_parameter={resource-id}]
+~~~
+
+DAP resource identifiers are always opaque byte strings, so anywhere
+`{resource-id}` occurs in a template (e.g., `{task-id}` or `{report-id}`) MUST
+be expanded to the URL-safe, unpadded Base 64 representation of the resource's
+identifier, as specified in sections 5 and 3.2 of {{!RFC4648}}.
```

Comment:
[RFC 6570](https://datatracker.ietf.org/doc/rfc6570/) discusses URI templating, and I considered using it here, but decided not to because (1) we don't need a template scheme as complicated as the one in 6570 and (2) 6570 has rules about how variables in templates should be percent encoded to be URL safe, whereas we have already settled on using URL-safe unpadded base64.